src/Microsoft.DotNet.Build.CloudTestTasks/ListAzureContainers.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Xml;
+
+using Task = Microsoft.Build.Utilities.Task;
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    public sealed class ListAzureContainers : Task
+    {
+        /// <summary>
+        /// The Azure account name used when creating the connection string.
+        /// </summary>
+        [Required]
+        public string AccountName { get; set; }
+
+        /// <summary>
+        /// The Azure account key used when creating the connection string.
+        /// </summary>
+        [Required]
+        public string AccountKey { get; set; }
+
+        /// <summary>
+         /// Prefix of Azure containers desired to return;
+         /// </summary>
+         public string Prefix { get; set; }
+
+        /// <summary>
+        /// An item group of blob filenames to download.  
+        /// </summary>
+        [Output]
+        public ITaskItem[] ContainerNames { get; set; }
+
+        public override bool Execute()
+        {
+            return ExecuteAsync().GetAwaiter().GetResult();
+        }
+
+        public async Task<bool> ExecuteAsync()
+        {
+            Log.LogMessage(MessageImportance.High, "List of Azure containers in storage account '{0}'.", AccountName);
```

Comment:
I would downgrade the importance level for the messages in this task, this is a utility task and the info is more useful as diagnostic output than in the typical case.